code/game/objects/pool/pool.dm

```diff
@@ -0,0 +1,308 @@
+	else
+		to_chat(user, "<span class='notice'>You start climbing down into [src]...")
+	if(do_after(user, 4 SECONDS, target=src))
+		splash(dropping)
+
+/datum/mood_event/poolparty
+	description = "<span class='nicegreen'>I love swimming!.</span>\n"
+	mood_change = 2
+	timeout = 2 MINUTES
+
+/datum/mood_event/robotpool
+	description = "<span class='warning'>I really wasn't built with water resistance in mind...</span>\n"
+	mood_change = -3
+	timeout = 2 MINUTES
+
+/datum/mood_event/poolwet
+	description = "<span class='warning'>Eugh! my clothes are soaking wet from that swim.</span>\n"
+	mood_change = -4
+	timeout = 4 MINUTES
+
+/turf/open/indestructible/sound/pool/proc/splash(mob/user)
+	user.forceMove(src)
+	playsound(src, 'sound/effects/splosh.ogg', 100, 1) //Credit to hippiestation for this sound file!
+	user.visible_message("<span class='boldwarning'>SPLASH!</span>")
+	var/zap = FALSE
+	if(issilicon(user)) //Do not throw brick in a pool. Brick begs.
+		zap = TRUE //Sorry borgs! Swimming will come at a cost.
+	if(ishuman(user))
+		var/mob/living/carbon/human/F = user
+		var/datum/species/SS = F.dna.species
+		if(MOB_ROBOTIC in SS.inherent_biotypes)  //ZAP goes the IPC!
+			zap = 2 //You can protect yourself from water damage with thick clothing.
+		if(F.head && istype(F.head, /obj/item/clothing))
+			var/obj/item/clothing/CH = F.head
+			if (CH.clothing_flags & THICKMATERIAL) //Skinsuit should suffice! But IPCs are robots and probably not water-sealed.
+				zap --
+		if(F.wear_suit && istype(F.wear_suit, /obj/item/clothing))
+			var/obj/item/clothing/CS = F.wear_suit
+			if (CS.clothing_flags & THICKMATERIAL)
+				zap --
+	if(zap > 0)
+		user.emp_act(zap)
+		user.emote("scream") //Chad coders use M.say("*scream")
+		do_sparks(zap, TRUE, user)
+		to_chat(user, "<span class='userdanger'>WARNING: WATER DAMAGE DETECTED!</span>")
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "robotpool", /datum/mood_event/robotpool)
+	else
+		if(!check_clothes(user))
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolparty)
+			return
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolwet)
+
+//Largely a copypaste from shower.dm. Checks if the mob was stupid enough to enter a pool fully clothed. We allow masks as to not discriminate against clown and mime players.
+/turf/open/indestructible/sound/pool/proc/check_clothes(mob/living/carbon/human/H)
+	if(!istype(H)) //Don't care about non humans.
+		return FALSE
+	if(H.wear_suit && (H.wear_suit.clothing_flags & SHOWEROKAY))
+		// Do not check underclothing if the over-suit is suitable.
+		// This stops people feeling dumb if they're showering
+		// with a radiation suit on.
+		return FALSE
+
+	. = FALSE
+	if(H.wear_suit && !(H.wear_suit.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.w_uniform && !(H.w_uniform.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.head && !(H.head.clothing_flags & SHOWEROKAY))
+		. = TRUE
```

Comment:
This feels like it should return true if the clothes are ok, not if they are not ok